src/Compilers/CSharp/Portable/Lowering/LocalRewriter/DelegateCreationRewriter.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+/// <summary>
+/// This type helps rewrite the delegate creations that target static method groups to use a cached instance of delegate.
+/// </summary>
+internal sealed partial class DelegateCreationRewriter
```

Comment:
>DelegateCreationRewriter [](http://example.com/codeflow?start=30&length=24)

Renaming a new file while PR is in under review was a "bad" move. This creates problems for reviewers. In particular, the tool that we use for review no longer considers the old file and the new file related, we cannot perform an incremental review and pretty much are forced to review from scratch. Please temporarily rename the files back to their original name. Hopefully that will enable the ability to perform an incremental review. The files can be renamed at the end, once the review process is complete.

Reply:
OK, reverted.

Reply:
> OK, reverted.

Thanks! Renaming files back worked.